aggregator/leader_flush_mgr.go

```diff
@@ -282,20 +322,43 @@ func (t *leaderFlushTask) Run() {
 		go mgr.persistFlushTimes(t.flushTimes)
 	}
 
-	if t.shouldFlush {
-		var wgWorkers sync.WaitGroup
-		start := mgr.nowFn()
-		for _, flusher := range t.flushers {
-			flusher := flusher
-			wgWorkers.Add(1)
-			mgr.workers.Go(func() {
-				flusher.Flush()
-				wgWorkers.Done()
-			})
+	if !t.shouldFlush {
+		return
+	}
+
+	// If the instance is not found in the placement, this means the instance is no longer
+	// considered as part of the aggregation cluster and as such we should stop flushing.
+	shards, err := mgr.ownedShards()
+	if err == errInstanceNotFoundInPlacement {
+		return
+	}
+
+	var (
+		wgWorkers sync.WaitGroup
+		start     = mgr.nowFn()
+	)
+	for _, flusher := range t.flushers {
+		// By default traffic is cut off from a shard, unless the shard is in the list of
+		// shards owned by the instance, in which case the cutover time and the cutoff time
+		// are set to the corresponding cutover and cutoff times of the shard.
+		var cutoverNanos, cutoffNanos int64
+		if shards != nil {
```

Comment:
could shards ever be nil?

Reply:
Yeah could probably simplify it be handling errors earlier, will do that.